opencog/guile/SchemeSmob.cc

```diff
@@ -224,40 +224,12 @@ void SchemeSmob::module_init(void*)
 {
 	// The portion of (opencog) done in C++
 	register_procs();
-
-	// Set the library load path, so that other modules can find
-	// thier libraries. Copied from `scm/opencog.scm` and should stay
-	// in sync with that file.  This is NOT needed for ordinary usage
-	// from the guile REPL, but is needed by the unit tests.  The problem
-	// is that the unit tests create SchemeEval class directly, which
-	// causes this code here to run, which defines the opencog scheme
-	// module. Thus, a later `(use-modules opencog)` is a no-op because
-	// guile thinks that it already has done this. But it really hasn't;
-	// the contents of `opencog.scm` were never actually run. So what
-	// we do is to manually run the contents of that file, below.
-	// Something more elegant would be nice.
-	//
-	// lib64 is used by various versions of CentOS
-	scm_c_eval_string(
-		"(define path \"/usr/lib/opencog:/usr/lib64/opencog:/usr/local/lib/opencog:/usr/local/lib64/opencog\")");
-
-	scm_c_eval_string(
-		"(setenv \"LTDL_LIBRARY_PATH\""
-		"   (if (getenv \"LTDL_LIBRARY_PATH\")"
-		"      (string-append (getenv \"LTDL_LIBRARY_PATH\") \":\" path)"
-		"      path))");
-
-#define DO_THE_UBER_BAD_HACKERY_FOR_EFFING_UNIT_TESTS_GRRRR
-#ifdef DO_THE_UBER_BAD_HACKERY_FOR_EFFING_UNIT_TESTS_GRRRR
-	// Loading files from the project directory is broken by design.
-	// However, the unit tests are broken by design.
-	// We REALLY should not do this, it violates basic laws of security,
-	// usability, debuggability. But some people think that's OK.
-	// Too lazy to fix.  See issue
-	// https://github.com/opencog/atomspace/issues/705 for details.
-	scm_c_eval_string("(add-to-load-path \"" PROJECT_SOURCE_DIR "/opencog/scm\")");
-	scm_c_eval_string("(add-to-load-path \"" PROJECT_BINARY_DIR "\")");
-#endif
+	// For testing - to avoid requiring manual change of guile load path.
+	// Alternatives to always adding this path would be:
+	// - make ctest manage the environment variable GUILE_LOAD_PATH during testing
+	// - set GUILE_LOAD_PATH env var in each test, before initialising Scheme
+	// - pass an optional load path through SchemeEval/SchemeSmob
+	scm_c_eval_string("(add-to-load-path \"" PROJECT_BINARY_DIR "/opencog/scm\")");
```

Comment:
well, this is still hackalicious, and should be marked as such ... 

The first alternative seems best.   Or even the second -- about half the unit tests already contain lines similar to `scm_c_eval_string("(add-to-load-path \"../../..\")")` trying to manually fiddle their way back to the PROJECT_SOURCE_DIR and all of those hacks have been fragile and fiddly, because they would break at the drop of a hat (err, at the drop of some new theory about how unit tests should work, imposed by external forces I have no control over.   It would be nice to strip out all of that crud .. or make external forces pay the accumulated code debt from their forced ultimatums.)  Whatever. This is a step in the right direction, but its still a hack, and the comments should clearly state its a hack.  Like `XXX FIXME` hack.

Reply:
I clarified it's still a hack